service/src/worker.rs

```diff
@@ -84,35 +75,38 @@ where
 		}
 		Ok(())
 	}
+}
 
-	fn peers(&self) -> WorkerResult<Vec<EnclaveMetadata>> {
-		Ok(self.node_api.all_enclaves(None)?)
+/// Looks for new peers and updates them.
+pub trait UpdatePeers {
+	fn search_peers(&self) -> WorkerResult<Vec<Url>>;
+	fn store_peers(&mut self, peers: Vec<Url>) -> WorkerResult<()>;
+	fn update_peers(&mut self) -> WorkerResult<()> {
+		let peers = self.search_peers()?;
+		self.store_peers(peers)
 	}
 }
 
-/// Temporary method that transforms the workers rpc port of the direct api defined in rpc/direct_client
-/// to the new version in rpc/server. Remove this, when all the methods have been migrated to the new one
-/// in rpc/server.
-pub fn worker_url_into_async_rpc_url(url: &str) -> WorkerResult<String> {
-	// [Option("ws(s)"), //ip, port]
-	let mut url_vec: Vec<&str> = url.split(':').collect();
-	match url_vec.len() {
-		3 | 2 => (),
-		_ => return Err(Error::Custom("Invalid worker url format".into())),
-	};
-
-	let ip = if url_vec.len() == 3 {
-		format!("{}:{}", url_vec.remove(0), url_vec.remove(0))
-	} else {
-		url_vec.remove(0).into()
-	};
-
-	let port: i32 =
-		url_vec.remove(0).parse().map_err(|e: ParseIntError| Error::Custom(e.into()))?;
+impl<NodeApi, Enclave> UpdatePeers for Worker<Config, NodeApi, Enclave>
+where
+	NodeApi: PalletTeerexApi + Send + Sync,
+{
+	fn search_peers(&self) -> WorkerResult<Vec<String>> {
+		let enclaves = self.node_api.all_enclaves(None)?;
+		let mut peer_urls = Vec::<String>::new();
+		for enclave in enclaves {
+			// FIXME: This is temporary only, as block gossiping should be moved to trusted ws server.
+			let worker_api_direct = DirectWorkerApi::new(enclave.url);
+			peer_urls.push(worker_api_direct.get_untrusted_worker_url()?);
+		}
+		Ok(peer_urls)
+	}
 
-	Ok(format!("{}:{}", ip, (port + 1)))
+	fn store_peers(&mut self, peers: Vec<Url>) -> WorkerResult<()> {
```

Comment:
This is a plain setter. The method should be called accordingly: `set_peers()`

Reply:
Renamed